multitenant: add AdminScatterRequest capability

ecwall opened the pull request:
Fixes #98394

1) Add a new `tenantcapabilities.CanAdminScatter` capability.
2) Check capability in `Authorizer.HasCapabilityForBatch`.

Release note: None

ecwall committed:
multitenant: add AdminScatterRequest capability

Fixes #98394

1) Add a new `tenantcapabilities.CanAdminScatter` capability.
2) Check capability in `Authorizer.HasCapabilityForBatch`.

Release note: None

ecwall committed:
multitenant: multitenant_admin_function_test cleanup

Get cluster settings and capabilities directly from testCase instead of
passing through testClusterCfg.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/98395)
<!-- Reviewable:end -->

knz reviewed (approved):
nice

ecwall committed:
multitenant: reduce capability unrelated test churn

Exclude capabilities from `tenant` logic test, `restore-tenants` data driven
test. These tests do not set or depend on capabilities and they are tested
elsewhere.

Release note: None

ecwall committed:
multitenant: reduce capability unrelated test churn

Filter out static capabilities in `subtest node_tenant_capabilities_cache`.

Release note: None

ecwall commented:
bors r=knz

ecwall commented:
bors r=knz
